src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,179 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.InvalidUserException;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final Long VALID_ID = 1L;
+    private static final int VALID_AGE_MIN = 18;
+    private static final String VALID_LOGIN_MIN = "new111";
+    private static final String VALID_PASSWORD_MIN = "new111";
+    private static final int VALID_AGE_AVG = 50;
+    private static final String VALID_LOGIN_AVG = "new111new111new111new111";
+    private static final String VALID_PASSWORD_AVG = "new111new111new111new111";
+    private static final int INVALID_AGE_MAX = 17;
+    private static final String INVALID_LOGIN_MAX = "mate1";
+    private static final String INVALID_PASSWORD_MAX = "mate1";
+    private static final int INVALID_AGE_AVG = 13;
+    private static final String INVALID_LOGIN_AVG = "mate";
+    private static final String INVALID_PASSWORD_AVG = "mate";
+    private static final int INVALID_AGE_MIN = 0;
+    private static final String INVALID_LOGIN_MIN = "";
+    private static final String INVALID_PASSWORD_MIN = "";
+
+    private static RegistrationService registrationService;
+    private static User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        user = new User();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user.setId(VALID_ID);
+        user.setLogin(VALID_LOGIN_MIN);
+        user.setPassword(VALID_PASSWORD_MIN);
+        user.setAge(VALID_AGE_MIN);
+    }
+
+    @Test
+    void register_validUserEdgeCaseIsCreated_Ok() {
+        User actual = registrationService.register(user);
+        assertTrue(actual.equals(user));
+    }
+
+    @Test
+    void register_validUserNonEdgeCaseIsCreated_Ok() {
```

Comment:
The word after underscore (_) should start in lowercase. Fix fir all names 
```suggestion
    void register_validUserNonEdgeCase_ok() {
```